docs/swift/DesignDoc.md

```diff
@@ -224,6 +245,16 @@ drafted or sent proposals for review that would benefit our designs:
 *   [Drafted: Resettable Properties]
     (https://github.com/patters/swift-evolution/blob/master/proposals/0000-resettable-properties.md)
 
+Lastly, we note some name transformations that we would apply to fields:
+
+* Repeated fields will have `Array` appended to their names. This is because
+  there is a trend in some messages to name repeated fields with the singular
+  version of the name (`descriptor.proto` in particular does this), and
+  `foo.name.append(bar)` looks potentially like appending a string to another
+  string whereas `foo.nameArray.append(bar)` is much clearer.
+* If a field name would collide with one of the properties or methods in
+  `ProtoMessage`'s public interface, we append `Field` to its name.
```

Comment:
just appending Array and Field mean we could have collisions:

```
optional int32 foo = 1;
repeated int64 foo_array = 2;
optional int32 int32 = 3;
optional int32 int32_field;
```

So you might want to stick underscores before what you append.  Same goes for appending Message/Enum to those types.